daprblog/content/posts/2023/dapr_flomesh/index.md

```diff
@@ -0,0 +1,546 @@
+
+[Flomesh](https://flomesh.io) service mesh uses the programmable proxy [Pipy](https://github.com/flomesh-io/pipy) as the core to provide east-west and north-south traffic management. Through L7-based traffic management capabilities, it breaks through the network isolation between computing environments, proposes a virtual flat network, and enables communication between applications in different computing environments. Flomesh service mesh can be imagined as a "big mesh" that covers multiple clusters.
+
+{{< imgproc flomesh_dapr_blog_arch.png  Resize "800x" />}}
+
+## Demo 
+
+The image below shows an example of how Dapr and Flomesh service mesh can be integrated to enable cross-cluster service communication. The example consists of two components: a server-side NodeApp and a client-side curl.
+
+The server-side NodeApp is a Dapr application, which is modified based on the NodeApp in the [Dapr hello-kubernetes](https://github.com/dapr/quickstarts/tree/master/tutorials/hello-kubernetes) example. When it returns a response, it displays the name of the current cluster.
+
+The client-side curl is used to send requests to NodeApp but is not declared as a Dapr application. This demonstrates how Flomesh service mesh can enable communication between a Dapr application and a non-Dapr application, enabling cross-cluster communication.
+
+
+{{< imgproc flomesh_dapr_blog_demo_arch.png  Resize "1200x" />}}
+
+
+The NodeApp in the example has three endpoints:
+
+- `GET /ports`: returns the ports that the application can be accessed on
+- `POST /neworder`: creates a new order
+- `GET /order`: queries an order
+  
+The following demonstration will start from creating a cluster and step-by-step introduce the configuration of the environment, installation and configuration of various components, deployment of applications, and so on.
+
+## One-Click Installation Script
+
+We also provide a script for one-click installation and quick experience, eliminating the tedious configuration of the environment and components. You can visit [GitHub](https://github.com/addozhang/flomesh-dapr-demo) to get the script content.
+
+Before using this script, make sure that `Docker` and `kubectl` are already installed on the system. The script will check during runtime and install tools such as `k3d`, `helm`, `jq`, `pv`, etc.
+
+- `flomesh.sh` - provides no parameters, the script will create 4 clusters, complete the environment installation and configuration, and run the demo
+- `flomesh.sh -h` - prints help information
+- `flomesh.sh -i` - creates 4 clusters, completes environment installation and configuration
+- `flomesh.sh -d` - runs the demo
+- `flomesh.sh -r` - cleans up demo-related resources
+- `flomesh.sh -u` - deletes all clusters
+- 
```

Comment:
```suggestion

```

I also like using tables... jus personal preference. I find them easier to skim. If you feel like updating this list to a table, here's the markdown for it - otherwise, no worries!

```
| Command         |  Description |
| --------------- | ------------- |
| `flomesh.sh`    | Provides no parameters, the script will create 4 clusters, complete the environment installation and configuration, and run the demo |
| `flomesh.sh -h` | Prints help information |
| `flomesh.sh -i` | Creates 4 clusters, completes environment installation and configuration |
| `flomesh.sh -d` | Runs the demo  |
| `flomesh.sh -r` | Cleans up demo-related resources |
| `flomesh.sh -u` | Deletes all clusters |
```